File: services/go/test_service_service/check_service.go
package test_service_service

import "errors"
import  "cloud.google.com/go/civil"
import "github.com/google/uuid"
import "github.com/shopspring/decimal"

type ServiceCheck struct{}

func (service *ServiceCheck) CheckEmpty() (*CheckEmptyResponse, error) {
	return nil, errors.New("implementation has not added yet")
}
func (service *ServiceCheck) CheckQuery(pString string, pStringOpt *string, pStringArray []string, pDate civil.Date, pDateArray []civil.Date, pDatetime civil.DateTime, pInt int, pLong int64, pDecimal decimal.Decimal, pEnum Choice, pStringDefaulted string) (*CheckQueryResponse, error) {
	return nil, errors.New("implementation has not added yet")
}
func (service *ServiceCheck) CheckUrlParams(intUrl int64, stringUrl string, floatUrl float32, boolUrl bool, uuidUrl uuid.UUID, decimalUrl decimal.Decimal, dateUrl civil.Date) (*CheckUrlParamsResponse, error) {
	return nil, errors.New("implementation has not added yet")
}
func (service *ServiceCheck) CheckForbidden() (*CheckForbiddenResponse, error) {
	return nil, errors.New("implementation has not added yet")
}

File: services/ts-express-superstruct/check_service.ts
import * as models from './models'

export interface CheckQueryParams {
    p_string: string,
    p_string_opt?: string | undefined,
    p_string_array: string[],
    p_date: string,
    p_date_array: string[],
    p_datetime: Date,
    p_int: number,
    p_long: number,
    p_decimal: number,
    p_enum: models.Choice,
    p_string_defaulted: string,
}

export interface CheckUrlParamsParams {
    int_url: number,
    string_url: string,
    float_url: number,
    bool_url: boolean,
    uuid_url: string,
    decimal_url: number,
    date_url: string,
    enum_url: models.Choice,
}

export type CheckForbiddenResponse =
    | { status: "ok", data: models.Message }
    | { status: "forbidden" }

export type SameOperationNameResponse =
    | { status: "ok" }
    | { status: "forbidden" }

export interface CheckService {
    checkEmpty(): Promise<void>
    checkQuery(params: CheckQueryParams): Promise<void>
    checkUrlParams(params: CheckUrlParamsParams): Promise<void>
    checkForbidden(): Promise<CheckForbiddenResponse>
    sameOperationName(): Promise<SameOperationNameResponse>
}
